drivers/hmis/app/jobs/hmis/merge_clients_job.rb

```diff
@@ -298,7 +298,13 @@ def update_personal_id_foreign_keys
 
     def destroy_merged_clients
       Rails.logger.info 'soft-deleting merged clients'
-      clients_needing_reference_updates.map(&:reload).map(&:destroy!)
+      ids = clients_needing_reference_updates.map(&:id)
+      scope = Hmis::Hud::Client.where(id: ids)
+      # preload associations to reduce n+1 when destroying a batch
+      preloads = Hmis::Hud::Client.reflect_on_all_associations.
+        filter { |a| a.options[:dependent] == :destroy }.
+        map(&:name)
+      scope.preload(*preloads).each(&:destroy!)
```

Comment:
When destroying a large number of records, the preload shaves off about 50% of the total queries (at least my simple test case). Note that we don't need to deeply descend into enrollments as those should already have been disassociated from the merged clients